src/Messages/TSMessagesManager.m

```diff
@@ -336,19 +336,6 @@ - (void)handleSecureMessageAsync:(OWSSignalServiceProtosEnvelope *)messageEnvelo
         NSString *recipientId = messageEnvelope.source;
         int deviceId = messageEnvelope.sourceDevice;
         dispatch_async([OWSDispatch sessionStoreQueue], ^{
-            if (![storageManager containsSession:recipientId deviceId:deviceId]) {
```

Comment:
This method checks if the *current* session is initialized, but what we really care about is if we have a session which can decrypt the method. This behavior is already implemented within session cipher.

So this code is sometimes overzealous, and in the case that it's not overzealous, it's redundant.